tests/test_transmission_matrix.py

```diff
@@ -328,6 +328,202 @@ def test_create_transmission_line_frequency_matching():
         ValueError, match="The frequencies do not match"):
         TransmissionMatrix.create_transmission_line(kl, Z)
 
+def test_create_conical_horn_number():
+    """Test `create_transmission_line` with impedance as number."""
+    k = FrequencyData([1j, 2, 3j], [1, 2, 3])
+    Z = 4+0.1j
+    
+    a = 0.3
+    b = 0.5
+    Omega = 0.4
+    
+    tmat_backwards = TransmissionMatrix.create_conical_horn(a, b, Omega, k, Z)
+    tmat_forwards = TransmissionMatrix.create_conical_horn(b, a, Omega, k, Z)
+
+    A = b/a * np.cos(k.freq*(b-a)) - 1/(k.freq*a) * np.sin(k.freq*(b-a))
+    B = 1j * Z / (a*b*Omega) * np.sin(k.freq*(b-a))
+    C = 1j * Omega / (k.freq*k.freq*Z) * ((1 + k.freq*k.freq*a*b) * np.sin(k.freq*(b-a)) - k.freq*(b-a)*np.cos(k.freq*(b-a)))
+    D = a/b * np.cos(k.freq*(b-a)) + 1/(k.freq*b) * np.sin(k.freq*(b-a))
```

Comment:
```suggestion
    A_desired = 3
    B_desired = 7+1j
    C_desired = 3
    D_desired = 42
```

maybe for testing it would be better to calcualte it for inputs